client/state/post-formats/reducer.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { isValidStateWithSchema } from 'state/utils';
+import * as schema from './schema';
+import {
+	SERIALIZE,
+	DESERIALIZE,
+	POST_FORMATS_RECEIVE,
+	POST_FORMATS_REQUEST,
+	POST_FORMATS_REQUEST_SUCCESS,
+	POST_FORMATS_REQUEST_FAILURE
+} from 'state/action-types';
+
+/**
+ * Returns the updated requests state after an action has been dispatched. The
+ * state maps site ID keys to whether a request for post formats is in progress.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function requesting( state = {}, action ) {
```

Comment:
The [`createReducer` utility](https://github.com/Automattic/wp-calypso/blob/997ace5791dfb6564a85d8f03722581addf32b24/client/state/utils.js#L44-L118) can be handy to avoid some of the boilerplate associated with schema validation in reducers. By default, if there's no schema defined, it will always return initial state, else perform the validation for you.